nifi-stateless/nifi-stateless-bundle/nifi-stateless-engine/src/main/java/org/apache/nifi/stateless/flow/StandardStatelessFlow.java

```diff
@@ -521,6 +522,26 @@ private void executeDataflow(final BlockingQueue<TriggerResult> resultQueue, fin
         }
     }
 
+    @Override
+    public boolean isStateful() {
+        return isStateful(rootGroup);
+    }
+
+    private boolean isStateful(final ProcessGroup processGroup) {
+        boolean isStateful = processGroup.getProcessors().stream()
+                .filter(processorNode -> processorNode.getProcessor().getClass().getAnnotation(Stateful.class) != null)
+                .findAny()
+                .isPresent();
+
+        if (isStateful) {
+            return true;
+        }
+
+        return processGroup.getProcessGroups().stream()
+                .filter(childGroup -> isStateful(childGroup))
+                .findAny()
+                .isPresent();
```

Comment:
Think we can simplify this in much the same way:
```
return processGroup.getProcessGroups().stream()
    .anyMatch(this::isStateful);
```

Reply:
Nice